### ISSUE TITLE
Contradicting statements in rlhf loss computation

### ISSUE THREAD

**Issue by jboboyle** [CarperAI/trlx#246]
### 🐛 Describe the bug

The inline comment [(line 63)](https://github.com/CarperAI/trlx/blob/dcbf7b0a37c1a1a49e5171338723cd19dd3f61e4/examples/summarize_rlhf/reward_model/reward_model.py#L63)  of  reward_model.py states "Compute pairwise loss. Only backprop on the last value before padding". However, when the loss is computed in [line 94](https://github.com/CarperAI/trlx/blob/dcbf7b0a37c1a1a49e5171338723cd19dd3f61e4/examples/summarize_rlhf/reward_model/reward_model.py#L94), CE loss is computed using the chosen and rejected reward segments truncated between the divergence index and index of the last value before padding. Should the loss be computed using only the chosen_end_scores and rejected_end_scores, or is the inline comment incorrect? 

### Which trlX version are you using?

`trlx==0.4`

**Comment by jon-tow** [comment]
CC @PhungVanDuy who might have a better idea off the top of their head.

**Comment by PhungVanDuy** [comment]
> 

Thank you for your comment, it is not pairwise like you said it is truncated segments from chosen and rejected. Sorry for the comment made you confused, we will remove it.

**Comment by jboboyle** [comment]
Thanks for the clarification! @PhungVanDuy - Do you know if that is standard practice for reward network training? Most HFRL papers I've seen for LM fine-tuning don't specify whether the reward network is trained on all tokens or only the last, and instead only mention that a scalar reward is assigned to each sample. I've seen a [repo](https://github.com/Dahoas/reward-modeling/blob/main/reward-modeling/finetune_rm.py) from @Dahoas that computes pairwise loss when training the reward network. Just wondering your thinking behind your approach and if you know how performance compares experimentally. Thank you!

**Comment by PhungVanDuy** [comment]
> 
`Most HFRL papers I've seen for LM fine-tuning don't specify whether the reward network is trained on all tokens or only the last, and instead only mention that a scalar reward is assigned to each sample.` --> I believe that are variations of the way to implement in the paper they will not talk more detail the idea is to re-use the trained SFT model with a linear head to get the scalar reward.

Actually, my implementation is the same as the implementation by Alex (Dahoas).In terms why I used the last token for the reward instead of the mean or something because I also tried to use the mean of tokens instead of just using the last token and the experiment result showed that the last token brings better performance.
